File: Beecrowd/Beginner/1435 - 1985/1435 - Square Matrix I/Python/1435_SquareMatrixI.py
while True:
    arraySize = int(input())
    if arraySize == 0:
        break

    matrix = [[0 for _ in range(arraySize)] for _ in range(arraySize)]
    counter = 1

    for i in range(arraySize):
        for j in range(i, arraySize - i):
            for k in range(i, arraySize - i):
                matrix[j][k] = counter

        counter += 1

    for i in range(arraySize):
        print(" ".join(f"{matrix[i][j]:>3}" for j in range(arraySize)))
    print()

File: Beecrowd/1 - Beginner/1435 - 1985/1435 - Square Matrix I/C++/1435_SquareMatrixI.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int arraySize;

    while (cin >> arraySize && arraySize != 0) {
        int matrix[arraySize][arraySize], counter = 1;

        for (int i = 0; i < arraySize; i++) {
            for (int j = i; j < arraySize - i; j++) {
                for (int k = i; k < arraySize - i; k++) {
                    matrix[j][k] = counter;
                }
            }

            counter++;
        }

        for (int i = 0; i < arraySize; i++) {
            for (int j = 0; j < arraySize - 1; j++) {
                cout << setw(3) << matrix[i][j] << " ";
            }

            cout << setw(3) << matrix[i][arraySize - 1] << endl;
        }

        cout << endl;
    }

    return 0;
}